lxd/storage/drivers/volume.go

```diff
@@ -74,6 +75,28 @@ func (v Volume) MountPath() string {
 	return GetVolumeMountPath(v.pool, v.volType, v.name)
 }
 
+// CreateMountPath creates the volume's mount path and sets the correct permission for the type.
+func (v Volume) CreateMountPath() error {
```

Comment:
@stgraber please can you take a close look at this function and check I haven't introduced any security flaws. Thanks

Reply:
Looks good to me